### PR TITLE
Simplify feature dependencies

### PR DESCRIPTION
**Pull request by thomaseizinger**
1. We only use `futures-util`'s `alloc` feature in the tests.
2. `futures-core` only has one feature and that is alloc and it is on by default.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,9 +15,9 @@ rust-version = "1.60.0"
 [dependencies]
 async-trait = "0.1.56"
 catty = "0.1.5"
-futures-core = { version = "0.3.21", default-features = false, features = ["alloc"] }
+futures-core = "0.3.21"
```
Comment (Restioson) [inline Cargo.toml:18]: Does futures core have no default features?

Comment (thomaseizinger) [reply Cargo.toml:18]: It does: `alloc` :)

**Comment by Restioson** [comment]
While this is a nice simplification, I am not sure if it is necessarily a good idea. It is a little less clear exactly what features are enabled and which aren't, as we are relying on what the default features are, as opposed to writing them out ourself. What do you think?

**Comment by thomaseizinger** [comment]
> While this is a nice simplification, I am not sure if it is necessarily a good idea. It is a little less clear exactly what features are enabled and which aren't, as we are relying on what the default features are, as opposed to writing them out ourself. What do you think?

I would expect `futures-core` to always be extremely lightweight so the risk of having more code "slip in" is quite low I believe. Plus, I'd expect any changes to features to result in a minor version bump as they can be breaking.

Happy to change it back but I also like the simplicity of not needing to set features. I do see though how it can trigger the thought  of "did we vet the default features of this lib?".

I wish default features would have never been introduced :(

**Comment by Restioson** [comment]
> I would expect futures-core to always be extremely lightweight so the risk of having more code "slip in" is quite low I believe. Plus, I'd expect any changes to features to result in a minor version bump as they can be breaking.

Fair enough.

> Happy to change it back but I also like the simplicity of not needing to set features. I do see though how it can trigger the thought of "did we vet the default features of this lib?".

Yea, this was my main point

**Comment by thomaseizinger** [comment]
> > Happy to change it back but I also like the simplicity of not needing to set features. I do see though how it can trigger the thought of "did we vet the default features of this lib?".
> 
> Yea, this was my main point

Do you want me to add a comment?

**Comment by Restioson** [comment]
I think that's a good solution, thanks :)

**Review by Restioson — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Resolved! Sorry for the delay.

**Pull request merged**